lib/src/kvs/tikv/mod.rs

```diff
@@ -68,23 +67,33 @@ impl Datastore {
 	}
 	/// Start a new transaction
 	pub(crate) async fn transaction(&self, write: bool, lock: bool) -> Result<Transaction, Error> {
+		// TiKV currently has issues with pessimistic locks. Panic in development.
+		#[cfg(debug_assertions)]
+		if lock {
+			panic!("There are issues with pessimistic locking in TiKV");
+		}
 		// Set whether this should be an optimistic or pessimistic transaction
 		let mut opt = if lock {
 			TransactionOptions::new_pessimistic()
 		} else {
 			TransactionOptions::new_optimistic()
 		};
 		// Set the behaviour when dropping an unfinished transaction
-		opt = opt.drop_check(CheckLevel::Panic);
+		opt = opt.drop_check(CheckLevel::Warn);
 		// Set this transaction as read only if possible
 		if !write {
 			opt = opt.read_only();
 		}
-		// Create a new distributed transaction
+		// Specify the check level
+		#[cfg(not(debug_assertions))]
+		let check = Check::Warn;
+		#[cfg(debug_assertions)]
+		let check = Check::Panic;
```

Comment:
Wondering if we will want this to be env though. There are likely to be cases where someone can reproduce but we can't. And then we could say "Please export SURREAL_PANIC_TX_UNCOMMITED=1".

Reply:
Yeah that's a good point. I'll leave that for a separate PR @phughk. Maybe let's add a separate issue just for that (as opposed to part of your bigger issue)?

Reply:
https://github.com/surrealdb/surrealdb/issues/2586